src/Microsoft.AspNetCore.Routing/Matching/ParameterPolicyFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Routing.Matching
+{
+    internal abstract class ParameterPolicyFactory
+    {
+        public abstract IParameterPolicy Create(string parameterName, string value, bool optional);
```

Comment:
I'm wondering if we want to have all of these methods take `RoutePatternParameterPart`. If we add more stuff to parameters in the future, it will be there.

Reply:
It is an internal type. Does it matter that much?

Or do you foresee the factory becoming public?

Reply:
I think we were talking about tusing this from MVC during the building of endpoints right?

Reply:
Good point.

I'm not a big fan of coupling it to patterns. I do think this class + abstract base class should be moved to Microsoft.AspNetCore.Routing namespace though.

Reply:
Derp, it is already coupled to patterns.